schemas/styles/csl-citation-label-format.rnc

```diff
@@ -0,0 +1,101 @@
+namespace a = "http://relaxng.org/ns/compatibility/annotations/1.0"
+namespace cs = "http://purl.org/net/xbiblio/csl"
+
+
+## cs:citation-label-format - Format for automatically generated citation-label
+div {
+  style.citation-label-format =
+    
+    ## Specify the format for automatically-generated citation-labels.
+    ## If unspecified, citation-labels are not automatically generated.
+    element cs:citation-label-format {
+      (
+       ## Specify a predefined label style
+       attribute label-style {
+         
+         ## Style used in citeproc-js for CLS 1.0.* styles
+         "citeproc"
+         | 
+           ## Default style used in biblatex
+           "biblatex-default"
+       }
+       | 
+         ## Specify a custom label style
+         (rendering-element+, citation-label.options)),
+      affixes,
+      [ a:defaultValue = "" ] delimiter,
+      font-formatting
+    }
+  
+  ## Citation Label Options
+  citation-label.options =
+    
+    ## Default values for inheritable attributes in citation-label-format
+    ## [ a:defaultValue = "" ] names-delimiter
+    ## [ a:defaultValue = "" ] name-delimiter
+    ## [ a:defaultValue = "short" ] name-form
+    names-inheritable-options,
+    name-inheritable-options,
+    
+    ## Abbreviate non-date variables by extracting the first N characters.
+    ## For list variables (e.g. names), extract chacters from each element.
+    attribute width { xsd:integer }?,
```

Comment:
If this is optional, and there's no default, what happens if it's missing? The complete name is rendered?

Reply:
Yes. If width is empty, no abbreviation is applied.